scripts/onboarding/aks/onboarding-msi-terraform-syslog/main.tf

```diff
@@ -0,0 +1,75 @@
+resource "azurerm_resource_group" "rg" {
+  name     = var.aks_resource_group_name
+  location = var.resource_group_location
+}
+
+resource "azurerm_kubernetes_cluster" "k8s" {
+  name                = var.cluster_name
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  dns_prefix          = var.dns_prefix
+
+  tags = var.resource_tag_values
+  
+  default_node_pool {
+    name       = "agentpool"
+    vm_size    = "Standard_D2_v2"
+    node_count = var.agent_count
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  oms_agent {
+    log_analytics_workspace_id = var.workspace_resource_id
+    msi_auth_for_monitoring_enabled = true
+  }
+
+  network_profile {
```

Comment:
expose the network profile settings to the variables.tf so that customer can use the network profile settings which matches their requirements.

Reply:
removed since if the value set other than kubenet, other fields need to be set too, user can add if reference this guide: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#network_plugin